lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1207,12 +1207,60 @@ def configure_disks(self, vm_obj):
                 elif disk_type == 'eagerzeroedthick':
                     diskspec.device.backing.eagerlyScrub = True
 
-            # which datastore?
+            # Select actual datastore and datastore name while creating data disk for
+            # virtual machine.
+            if vm_obj is None:
+                vm_name = self.params['name'].replace('.', '_')
+            else:
+                vm_name = vm_obj.name
+
             if expected_disk_spec.get('datastore'):
-                # TODO: This is already handled by the relocation spec,
-                # but it needs to eventually be handled for all the
-                # other disks defined
-                pass
+                # User has specified datastore name
+                datastore_name = expected_disk_spec['datastore']
+                datastore = self.cache.find_obj(self.content, [vim.Datastore], datastore_name)
+
+                if datastore is None:
+                    self.module.fail_json(msg="Unable to find datastore named %s specified "
+                                              "in disk configuration at index %s" % (datastore_name,
+                                                                                     disk_index))
+                parent_dc = self.cache.get_parent_datacenter(datastore)
+                vmdk_file_name = self.create_directory_for_datadisk(datacenter=parent_dc,
+                                                                    datastore=datastore,
+                                                                    vm_name=vm_name,
+                                                                    diskspec=diskspec)
+                diskspec.device.backing.datastore = datastore
+                diskspec.device.backing.fileName = vmdk_file_name
+
+            elif 'autoselect_datastore' in expected_disk_spec and expected_disk_spec['autoselect_datastore']:
```

Comment:
`if expect_disk_spec.get('autoselect_datastore', False):`

Reply:
Also, if the disk already exist, 'autoselect_datastore' shouldn't modify the disk.

Reply:
ok.